Dockerfile.template

```diff
@@ -0,0 +1,30 @@
+# base-image for node on any machine using a template variable,
+# see more about dockerfile templates here: https://www.balena.io/docs/learn/develop/dockerfile/#dockerfile-templates
+# and about balena base images here: https://www.balena.io/docs/reference/base-images/base-images/
+FROM balenalib/%%BALENA_MACHINE_NAME%%-node:latest
```

Comment:
A cool trick we can add is use multi-stage Dockerfile to build the TS project on a first stage and then copy over the build directory into a second slimmer stage. This is how the [supervisor](https://github.com/balena-io/balena-supervisor/blob/master/Dockerfile), albeit it's a rather complex Dockerfile to look at haha. [Here](https://github.com/balenalabs/balena-sound/blob/master/core/sound-supervisor/Dockerfile.template) is a simpler one that works too. 
We could add this as another separate `Dockerfile.production` and mention it in the README.

Reply:
I had thought of that for a bit, but then thought this should be a very simple hello world example, not really intended for production. So, we should probably keep it simple (even though not very optimized)? What I was not sure is if I should use `ts-node` or do the explicit two-phase `build` and `start` that I do now. Again, for _simplicity_, I opted for the latter and did not introduce another dependency.